tests/e2e/approval-diff.spec.ts

```
@@ -0,0 +1,314 @@
+        fs.writeFileSync(taskJsPath, original, 'utf8');
+        // Converge rather than approving once: the task is still pending at
+        // the PREVIOUS content when the bytes are restored, so a single
+        // approve would record that stale hash and leave the restored file
+        // permanently pending. Keep approving whatever is currently pending
+        // until the reconciler has caught up to the restored bytes and the
+        // task settles armed.
+        const deadline = Date.now() + 60_000;
+        for (;;) {
+          const t = await (await request.get(`/api/tasks/${encodeURIComponent(MANUAL_TASK_ID)}`)).json();
+          if (t.pending_approval !== true) break;
+          if (Date.now() > deadline) throw new Error('task did not settle after restore');
```

Biome (2.5.3)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)